src/progress.jl

```diff
@@ -0,0 +1,110 @@
+import Base: done
+
+export ProgressBar, progress!, msg!, done, right_text!, @progress
+
+type ProgressBar
+  leftText::String
+  rightText::String
+  id::String
+  progress::Float64
+  msg::String
+  determinate::Bool
+end
+
+"""
+    ProgressBar(;name = "", msg = "")
+
+Create a new progress bar and register it with Juno, if possible.
+"""
+function ProgressBar(;name = "", msg = "")
+  id = randstring(10)
```

Comment:
Yeah, I wanted to do that but was to lazy to figure out how to get them (since they aren't exported from Base).